From 862029ae5315db204af3b35fc12acaa4ca12c50f Mon Sep 17 00:00:00 2001
From: Daniel Kurtz <djkurtz@chromium.org>
Date: Fri, 13 May 2011 18:59:09 +0800
Subject: [PATCH] wireless-power tries to control wifi even if iwconfig is not found

When iwconfig isn't found, wireless-power still tries to execute
"$IWCONFIG $IFNAME", which fails with "$IFNAME not found" errors.

BUG=chromium-os:15185
TEST=/usr/sbin/laptop_mode auto force
There should be no more errors like this:
/usr/share/laptop-mode-tools/modules/wireless-power: 94: eth0: not found Failed.
/usr/share/laptop-mode-tools/modules/wireless-power: 94: usb0: not found Failed.
/usr/share/laptop-mode-tools/modules/wireless-power: 94: wlan0: not found Failed
---
 usr/share/laptop-mode-tools/modules/wireless-power |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/usr/share/laptop-mode-tools/modules/wireless-power b/usr/share/laptop-mode-tools/modules/wireless-power
index 6407d2a..649b40a 100755
--- laptop-mode-tools-1.59/usr/share/laptop-mode-tools/modules/wireless-power
+++ laptop-mode-tools-1.59/usr/share/laptop-mode-tools/modules/wireless-power
@@ -62,6 +62,7 @@ if [ x$CONTROL_WIRELESS_POWER_SAVING = x1 ] || [ x$ENABLE_AUTO_MODULES = x1 -a x
 		IWCONFIG=/usr/sbin/iwconfig
 	else
 		log "VERBOSE" "iwconfig is not installed"
+		exit 1
 	fi
 
 	# Translate 1 => on, 0 => off
-- 
1.7.3.1

